### PR TITLE
Feat(next.js): Migration to next.js

### PR DESCRIPTION
**Pull request by AugustinMauroy**
# Introduce Next.js

Sadly, it's not really the same in style. but in content it's same.

After merge of this pr I will improve the project.

> don't worry!

Implement next.js help to maintain the website

# feature added

* Typescript support
* Twitter link
* Discord link
* Icon for link
* Dark mode
* Linter
* Code format
* 404 page
* Sass support

# feature can be add in futur

> It's not for this pr it's for after merging 

* I18n support
* Markdown support 
* Unit test

# Related Issue

#67

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
+  "description": "PrismarineJS website build with Next.js",
   "private": true,
   "homepage": "https://prismarinejs.github.io/",
   "dependencies": {
-    "bootstrap": "^5.0.1",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "react-router-dom": "^5.2.0",
-    "react-scripts": "^3.4.1"
+    "next": "latest",
```
Comment (rom1504) [inline package.json:8]: That's a bad practice, use Carret versions

**Comment by rom1504** [comment]
CI failing

File: package.json (hunk 1 above)
Comment (AugustinMauroy) [reply package.json:8]: Yep I will change this

**Comment by AugustinMauroy** [comment]
> CI failing

MMM it's about telemetry data. What do you thinks?

**Comment by AugustinMauroy** [comment]
To merge I propose to use squash and merge

**Comment by AugustinMauroy** [comment]
I don't find solution to fix the security.

File: components/Footer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { FaTwitter, FaDiscord, FaGithub } from 'react-icons/fa'
+import Styles from './index.module.css'
+
+export default function Footer(): JSX.Element {
+    return (
+        <div className={Styles.social}>
+            <Link href='https://twitter.com/prismarinejs'>
+                <FaTwitter className={Styles.icon} />
+            </Link>
+            <Link href=''>
```
Comment (AugustinMauroy) [inline components/Footer/index.tsx:11]: Can you give me the correct link ?

**Comment by AugustinMauroy** [comment]
Review is ready !

**Comment by rom1504** [comment]
CI failing

**Comment by rom1504** [comment]
This doesn't seem to bring any new features
What's the plan on this ?

**Comment by AugustinMauroy** [comment]
> CI failing

IDK why? Because I have add .env with telemetry at off and add socket file to with ignore the package.
So idk.

**Comment by AugustinMauroy** [comment]
> This doesn't seem to bring any new features
> What's the plan on this ?


Technically it's just the migration.

* In long terme I will add one page for one project.
* Support I18n
* Add blog/post (for example: realease note)

**Comment by AugustinMauroy** [comment]
They are feat added:


* discord link
* twitter link

**Comment by AugustinMauroy** [comment]
@SocketSecurity ignore next@12.3.4

**Comment by AugustinMauroy** [comment]
@rom1504

Can you do the last review ?

And do this `@SocketSecurity ignore next@13.1.6`

**Comment by rom1504** [comment]
blocked on adding any new feature (can be done in other PRs)

**Comment by rom1504** [comment]
this increases the complexity

I think we should rather decrease it

**Comment by AugustinMauroy** [comment]
It increases the complexity a bit. But it also increases the DX and stability.

**Comment by AugustinMauroy** [comment]
It increases the complexity a bit. But it also increases the DX and stability.

**Comment by AugustinMauroy** [comment]
Oups fail for closing

**Comment by rom1504** [comment]
apparently it was not clear in my previous messages: if there is no feature added, I think we should not move to next.js

**Review by Hardel-DW — APPROVED** [review]
It's Good

**Comment by andriycraft** [comment]
why was this closed

**Comment by AugustinMauroy** [comment]
https://github.com/PrismarineJS/prismarine-meta/issues/5

**Pull request closed without merging**